peekingduck/pipeline/nodes/output/screen.py

```diff
@@ -45,23 +47,77 @@ class Node(AbstractNode):
             with reference from the top left corner of the screen, in pixels.
     """
 
+    ASPECT_RATIO_THRESHOLD = 5e-3
+    MIN_SIZE = 120
+
     def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
         cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
         cv2.moveWindow(self.window_name, self.window_loc["x"], self.window_loc["y"])
+        self.current_filename = ""
 
     def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         """Show the outputs on your display"""
-
         img = inputs["img"]
-        if self.window_size["do_resizing"]:
-            img = cv2.resize(
-                img, (self.window_size["width"], self.window_size["height"])
-            )
+        self._set_window_size(inputs["filename"], img)
         cv2.imshow(self.window_name, img)
-
         if cv2.waitKey(1) & 0xFF == ord("q"):
             cv2.destroyWindow(self.window_name)
             return {"pipeline_end": True}
 
         return {"pipeline_end": False}
+
+    def _set_window_size(self, input_filename: str, img: np.ndarray) -> None:
+        """If `do_resizing` option is False, window size will be initialized to the image or
+        video's frame default size. When the user changes the window's size, the aspect ratio
+        will be kept unchanged.
+
+        If `do_resizing` option is True, window size will be initialized to the config setting's
+        width and height for every new video or image. The aspect ratio will not be kept constant
+        when the user changes the size of the display window.
+
+        Args:
+            input_filename (str): The filename from the `inputs` dictionary
+            img (np.ndarray): The current image. The image will not be changed in this function.
+        """
+        if not self.window_size["do_resizing"]:
+            img_height, img_width, _ = img.shape
+            if input_filename != self.current_filename:
+                # Initialize the window size for every new video
+                cv2.resizeWindow(self.window_name, img_width, img_height)
+                self.current_filename = input_filename
+            else:
+                # Check the current window size, resize it if aspect ratio is not the same as image
+                aspect_ratio = img_width / img_height
+                _, _, win_width, win_height = cv2.getWindowImageRect(self.window_name)
+                if (
+                    abs(win_width / win_height - aspect_ratio)
+                    > self.ASPECT_RATIO_THRESHOLD
+                ):
+                    # Make sure window height and width is above lower bound
+                    win_width = (
+                        win_width if win_width > self.MIN_SIZE else self.MIN_SIZE
+                    )
```

Comment:
Consider using `win_width = max(win_width, self.MIN_SIZE)` instead to clearly show the intent of this statement

Reply:
amended